common/shared.c

```diff
@@ -68,44 +68,60 @@ const unsigned int test_cnt = (sizeof test_size / sizeof test_size[0]);
 static const char integ_alphabet[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 static const int integ_alphabet_length = (sizeof(integ_alphabet)/sizeof(*integ_alphabet)) - 1;
 
-static int getaddr(char *node, char *service, void **addr,
-			size_t *len)
+#define DUP_ADDR(_dst_addr, _dst_addrlen,			\
+		_src_addr, _src_addrlen, _ret)			\
+	do {							\
+		_dst_addr = malloc(_src_addrlen);		\
+		if (!_dst_addr) {				\
+			FT_ERR("address allocation failed\n");	\
+			_ret = EAI_MEMORY;			\
+		}						\
+		_dst_addrlen = _src_addrlen;			\
+		memcpy(_dst_addr, _src_addr, _dst_addrlen);	\
+	} while(0)
```

Comment:
Please just make a normal function, and let the compiler decide to inline or not.  This doesn't need to be a macro.